source/core/module_eventHelpers.js

```diff
@@ -80,7 +73,14 @@
             unRegisterForControlEvent = fn;
         };
 
-        publicAPI.eventHelpers.occurEvent = Module.eventHelpers.occurEvent = OccurEvent;
+        publicAPI.eventHelpers.occurEvent = Module.eventHelpers.occurEvent = function (eventOracleIndex, controlId, eventType, typeVisitor, eventDataTypeValueRef, eventData) {
+            // Allocate space for the event data using the type information passed in to occurEvent
+            var allocatedDataValueRef = Module.eggShell.allocateData(eventDataTypeValueRef.typeRef);
+            Module.eggShell.reflectOnValueRef(typeVisitor, allocatedDataValueRef, eventData);
```

Comment:
I thought we were going to go with the writeCallback function.

Reply:
@rajsite suggested using a visitor instead of using a writeCallback.

Reply:
Memo, Jared, and I synced up and decided to use a writeCallback(allocatedValueRef) function that is referenced in each invocation of occurEvent